### PR TITLE
Update our Android Template to use Hilt.

### PR DESCRIPTION
**Pull request by minhnimble**
## What happened 🤔
#51 : Remove the old usage of Dagger for Dependency Injection and replace it with Hilt - a best practice library for doing DI:
https://developer.android.com/training/dependency-injection/hilt-android

Also, update our Android Template to use the latest versions for some Hilt-related libraries:
- updated `android_support_version` from '1.1.0-alpha01' to '1.1.0'
- added `fragment_ktx_version` with version '1.2.0'
- updated `arch_component_version` from '2.1.0-alpha01' to '2.2.0-rc02'

## Insight 👀
This is the migration guide:
https://dagger.dev/hilt/migration-guide

It is fun to explore, they also have a sample project for good reference. 👍

## PoW (a.k.a Proof Of Work)💪
Build success with the new Hilt applied
Screen Shot 2020-09-18 at 18 20 11

### CODE REVIEW

File: app/src/main/java/com/nimbl3/ui/main/MainViewModel.kt [hunk 1]
```diff
@@ -10,11 +11,10 @@ import com.nimbl3.ui.main.data.Data
 import io.reactivex.Observable
 import io.reactivex.subjects.BehaviorSubject
 import io.reactivex.subjects.PublishSubject
-import javax.inject.Inject
 
 class MainViewModel
-@Inject constructor(private val repository: ApiRepository,
-                    private val schedulers: SchedulersProvider) : BaseViewModel(), Inputs, Outputs {
+@ViewModelInject constructor(private val repository: ApiRepository,
+                                 private val schedulers: SchedulersProvider) : BaseViewModel(), Inputs, Outputs {
```
Comment (sleepylee) [inline app/src/main/java/com/nimbl3/ui/main/MainViewModel.kt:17]: a bit weird format maybe you would need to recheck your IDE code format style? We should be using Kotlin default formatter

File: build.gradle [hunk 2]
```diff
@@ -12,18 +12,20 @@ buildscript {
 
         // Dependencies (Alphabet sorted)
         android_legacy_support_version = '1.0.0'
-        android_support_version     = '1.1.0-alpha01'
+        android_support_version     = '1.1.0'
         android_ktx_version         = '1.0.1'
+        fragment_ktx_version        = '1.2.0'
         android_material            = '1.1.0-alpha02'
-        arch_component_version      = '2.1.0-alpha01'
+        arch_component_version      = '2.2.0-rc02'
```
Comment (sleepylee) [inline build.gradle:19]: please put these work updates to the description of the PR as well 🙏 at first I thought there were only work related to hilt but apparently you also updated many component lib versions as well

**Review by sleepylee — APPROVED** [review]
some minor comments, 1 related to code format just to recheck

File: app/build.gradle [hunk 3]
```diff
@@ -65,6 +66,10 @@ android {
         targetCompatibility JavaVersion.VERSION_1_8
     }
 
+    kotlinOptions {
+        jvmTarget = "1.8"
```
Comment (ViktorArtemiev) [inline app/build.gradle:70]: would be great if we init the property by the following const instead 🤔 
    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_1_8.toString()
    }

File: app/src/main/java/com/nimbl3/ui/main/MainViewModel.kt (hunk 1 above)
Comment (minhnimble) [reply app/src/main/java/com/nimbl3/ui/main/MainViewModel.kt:17]: yep, let me take a look on my IDE, it is freshly installed 😅

File: app/build.gradle (hunk 3 above)
Comment (minhnimble) [reply app/build.gradle:70]: yep, I will update, thanks man 👍

File: build.gradle (hunk 2 above)
Comment (minhnimble) [reply build.gradle:19]: noted 👍

File: app/src/main/java/com/nimbl3/ui/second/SecondActivity.kt [hunk 4]
```diff
@@ -66,7 +63,7 @@ class SecondActivity : BaseActivity() {
 
     override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults)
-        onRequestPermissionsResult(requestCode, grantResults)
+        onRequestPermissionsResult(requestCode, permissions ,grantResults)
```
Comment (ViktorArtemiev) [inline app/src/main/java/com/nimbl3/ui/second/SecondActivity.kt:66]: small formatting issue `permissions ,` -> `permissions, `

**Review by ViktorArtemiev — APPROVED** [review]

File: app/src/main/java/com/nimbl3/ui/second/SecondActivity.kt (hunk 4 above)
Comment (sleepylee) [reply app/src/main/java/com/nimbl3/ui/second/SecondActivity.kt:66]: yup

**Review by sleepylee — APPROVED** [review]
small format issue before you can merge

File: app/src/main/java/com/nimbl3/ui/second/SecondActivity.kt (hunk 4 above)
Comment (minhnimble) [reply app/src/main/java/com/nimbl3/ui/second/SecondActivity.kt:66]: noted, thanks for spotting

**Review by sleepylee — APPROVED** [review]

File: app/src/main/java/com/nimbl3/di/modules/GlideModule.kt [hunk 5]
```diff
@@ -2,6 +2,8 @@ package com.nimbl3.di.modules
 
 import com.bumptech.glide.annotation.GlideModule
 import com.bumptech.glide.module.AppGlideModule
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ApplicationComponent
```
Comment (luongvo) [inline app/src/main/java/com/nimbl3/di/modules/GlideModule.kt:6]: How about these imports? I don't see their usages 🤔

File: app/src/main/java/com/nimbl3/ui/main/MainViewModel.kt [hunk 6]
```diff
@@ -10,11 +11,10 @@ import com.nimbl3.ui.main.data.Data
 import io.reactivex.Observable
 import io.reactivex.subjects.BehaviorSubject
 import io.reactivex.subjects.PublishSubject
-import javax.inject.Inject
 
 class MainViewModel
-@Inject constructor(private val repository: ApiRepository,
-                    private val schedulers: SchedulersProvider) : BaseViewModel(), Inputs, Outputs {
+@ViewModelInject constructor(private val repository: ApiRepository,
+                             private val schedulers: SchedulersProvider) : BaseViewModel(), Inputs, Outputs {
```
Comment (luongvo) [inline app/src/main/java/com/nimbl3/ui/main/MainViewModel.kt:17]: should we consider to reformat this block to
```
class MainViewModel @ViewModelInject constructor(
    private val repository: ApiRepository,
    private val schedulers: SchedulersProvider
) : BaseViewModel(), Inputs, Outputs {
```
🤔

File: app/src/main/java/com/nimbl3/ui/second/SecondViewModel.kt [hunk 7]
```diff
@@ -1,16 +1,15 @@
...
 import com.nimbl3.ui.main.Const
 import com.nimbl3.ui.main.data.Data
 import io.reactivex.Observable
 import io.reactivex.subjects.BehaviorSubject
-import timber.log.Timber
-import javax.inject.Inject
 
 class SecondViewModel
-@Inject constructor(private val schedulers: SchedulersProvider) : BaseViewModel(), Inputs, Outputs {
+@ViewModelInject constructor(private val schedulers: SchedulersProvider) : BaseViewModel(), Inputs, Outputs {
```
Comment (luongvo) [inline app/src/main/java/com/nimbl3/ui/second/SecondViewModel.kt:12]: same, for long class definition

File: app/src/main/java/com/nimbl3/di/modules/GlideModule.kt (hunk 5 above)
Comment (minhnimble) [reply app/src/main/java/com/nimbl3/di/modules/GlideModule.kt:6]: oh, nice catch, I forget to remove, have been testing it 😅

File: app/src/main/java/com/nimbl3/ui/main/MainViewModel.kt (hunk 6 above)
Comment (minhnimble) [reply app/src/main/java/com/nimbl3/ui/main/MainViewModel.kt:17]: got it, let me refactor it

File: app/src/main/java/com/nimbl3/ui/second/SecondViewModel.kt (hunk 7 above)
Comment (minhnimble) [reply app/src/main/java/com/nimbl3/ui/second/SecondViewModel.kt:12]: got it, let me refactor it

**Review by luongvo — APPROVED** [review]
all lgtm 👏

**Review by sleepylee — APPROVED** [review]

**Pull request merged**